### ISSUE TITLE
Null string-literals not working within Arrays

### ISSUE THREAD

**Issue by Pyppe** [tminglei/slick-pg#327]
A simple way to reproduce this would be to add `"NULL"` for `strList` in `PgArraySupportSuite`:
```scala
  val testRec1 = ArrayBean(33L, List(101, 102, 103), Buffer(1L, 3L, 5L, 7L), List(1,7), List("robert}; drop table students--", "NULL"),
    Some(Vector("str1", "str3", "", " ")), List(uuid1, uuid2), List(Institution(113)), None)
```

What happens is that `"NULL"` String-literal gets interpreted as an actual `null` within the DB.

**Comment by Pyppe** [comment]
It seems that `SimpleArrayUtils.mkString` utilizes the `{'...'}` notation for constructing arrays, where Postgres first has to interpret the statement as string and then convert to array.

Maybe it would be easier to use the ARRAY construction syntax: https://www.postgresql.org/docs/9.6/static/sql-expressions.html#SQL-SYNTAX-ARRAY-CONSTRUCTORS

**Comment by tminglei** [comment]
@Pyppe we can't use ARRAY construction syntax here.

Slick uses PreparedStatement to prepare sql, and any non-number parameter will be embedded as `'xxx'`.

**Comment by tminglei** [comment]
Fixed. Can you help check?

**Comment by Pyppe** [comment]
Well, that was fast! 👍

Looks good to me. Tried with different inputs, and all seemed to work OK:
```scala
List("robert}; drop table students--", "NULL", "0", "-", "\"", "null", "\"null\"",
     "", " ", "  ", "'", "'null'", "{}", "{'}", "{}}")
```

**Comment by Pyppe** [comment]
Any idea, when this will be released (maven)?

**Comment by tminglei** [comment]
Released slick-pg v0.14.6, pls use it.

**Comment by Pyppe** [comment]
Awesome! But how about `0.15.0-*`. Because we are currently using `0.15.0-M3`.

**Comment by tminglei** [comment]
Just released slick-pg v0.15.0-M4.
